pulsar-client-api/src/main/java/org/apache/pulsar/client/api/Schema.java

```diff
@@ -92,6 +92,15 @@ default boolean supportSchemaVersioning() {
         return false;
     }
 
+    /**
+     * Return an instance of this schema at the given version.
+     * @param schemaVersion
+     * @return the schema at that specific version
+     */
+    default Schema<?> atSchemaVersion(byte[] schemaVersion) {
+        return this;
```

Comment:
Is this ok if supportSchemaVersioning() == true and schema versions mismatch?
What's the contract for the case when schema at this version cannot be found? null, exception, whatever other schema version we find?

Reply:
good point, I am adding "throws SchemaSerializationException" (that is the common Schema related exception for problems with the registry)